doc/userguide/file-extraction/config-update.rst

```diff
@@ -0,0 +1,38 @@
+Update File-store v1 Configuration to V2
+========================================
+
+Given a file-store configuration like::
+
+  - file-store:
+      enabled: yes        # set to yes to enable
+      log-dir: files      # directory to store the files
+      force-magic: no     # force logging magic on all stored files
+      force-hash: [md5]   # force logging of md5 checksums
+      force-filestore: no # force storing of all files
+      stream-depth: 1mb   # reassemble 1mb into a stream, set to no to disable
+      waldo: file.waldo   # waldo file to store the file_id across runs
+      max-open-files: 0   # how many files to keep open (O means none)
+      write-meta: yes     # write a .meta file if set to yes
+      include-pid: yes    # include the pid in filenames if set to yes.
+
+the following changes will need to be made to convert to a V2 style configuration:
+
+* A ``version`` field must be set to 2.
```

Comment:
should it default to 2 if not set at all?

Reply:
Right now the only way to use v2 is by setting the version. If we'd skip this I fear the only result will be that we take the v1 config as v2.

Reply:
I had been wondering about what to default to as well. Ideally things should just work with the minimal amount of configuration.  But with this being the first release without v1, I think we should enforce the version for now.

Reply:
Valid point, we could error out in later versions if old settings are still set.

Reply:
`log-dir` was renamed to `dir`.  So that is something we could check.  But in reality I'd like:
```
- file-store:
    enabled: yes
```
to just work with reasonable default.

Anyways, I'm OK with forcing the version to be set for now, then relaxing it later on..

Reply:
We don't want to auto-convert since we don't want to reuse the `log-dir`/`dir`, right? So then I rather keep the requirement for now. We can remove it with 7 to shrink the default config. We should then still add some checks for v1 config, in case ppl go from 5->7.

So perhaps we do need some extra checks anyway, in case ppl just add a `version: 2` to their v1 config and expect it to work.

Reply:
So currently in this PR, just a warning is logged if version isn't 2.  So the filestore is disabled, but Suricata keeps running.  Should we be more harsh so its brought to the users attention sooner?